virtool/api/custom_json.py

```diff
@@ -44,28 +25,59 @@ def isoformat(obj: datetime.datetime) -> str:
     return obj.replace(tzinfo=datetime.timezone.utc).isoformat().replace("+00:00", "Z")
 
 
-def dumps(obj: object) -> str:
+def default(obj):
+    if issubclass(type(obj), BaseModel):
+        return obj.dict(by_alias=True)
+    raise TypeError
+
+
+def dumps(obj: object) -> bytes:
     """
     A wrapper for :func:`dumps` is able to encode datetime objects in input.
 
     Used as `dumps` argument for :func:`.json_response`.
 
     :param obj: a JSON-serializable object
-    :return: a JSON string
+    :return: a JSON string in bytes
 
     """
-    return json.dumps(obj, cls=CustomEncoder)
+    return orjson.dumps(
+        obj,
+        default=default,
+        option=orjson.OPT_NAIVE_UTC | orjson.OPT_UTC_Z,
+    )
 
 
-def pretty_dumps(obj: object) -> str:
+def pretty_dumps(obj: object) -> bytes:
     """
     A wrapper for :func:`json.dumps` that applies pretty formatting to the output.
 
     Sorts keys and adds indentation. Used as ``dumps`` argument for
     :func:`.json_response`.
 
     :param obj: a JSON-serializable object
-    :return: a JSON string
+    :return: a JSON string in bytes
+
+    """
+    return orjson.dumps(
+        obj,
+        default=default,
+        option=orjson.OPT_INDENT_2
+        | orjson.OPT_SORT_KEYS
+        | orjson.OPT_NAIVE_UTC
+        | orjson.OPT_UTC_Z,
+    )
 
+
+def pretty_orjson_serializer(obj) -> str:
+    """
+    Used by SQLAlchemy as they expect strings.
+    """
+    return pretty_dumps(obj).decode()
+
+
+def orjson_serializer(obj) -> str:
+    """
+    Used by SQLAlchemy as they expect strings.
     """
-    return json.dumps(obj, cls=CustomEncoder, indent=4, sort_keys=True)
+    return dumps(obj).decode()
```

Comment:
Can you explain why there are two of these?

Reply:
In pg/utils, the json_serializer was `pretty_dumps`. However, in pg/testing the serializer was `dumps`, so I made two to preserve that functionality.

Reply:
That seems backwards to me.

I think we should get rid of the pretty serializer. The use I see for pretty is that the JSONB looks good in a snapshot or during debugging. There are better ways to do that than storing pretty JSON in Postgres.

Agree?